src/Profiles.ts

```diff
@@ -857,6 +876,69 @@ export class Profiles {
         return profileManager;
     }
 
+    public async validateProfiles(theProfile: IProfileLoaded) {
+        let filteredProfile: IProfileValidation;
+        let profileStatus;
+        const getSessStatus = await ZoweExplorerApiRegister.getInstance().getCommonApi(theProfile);
+
+        // Filter profilesForValidation to check if the profile is already validated
+        this.profilesForValidation.filter((profile) => {
+            if (profile.name === theProfile.name) {
+                filteredProfile = {
+                    status: profile.status,
+                    name: profile.name
+                };
+            }
+        });
+
+        // If not yet validated, call getStatus and validate the profile
+        // status will be stored in profilesForValidation
+        if (filteredProfile === undefined) {
```

Comment:
Just curious what the expected behavior is if the value is `null`.
I understand it can only happen if someone adds additional logic or initializes the valuable to null, but just as a general rule (maybe) we can check for `variable != null`. That way we cover both, undefined and null cases.

No changes requested though.

Reply:
@zFernand0 I'll take note of this! Thank you :)